src/pages/Home.jsx

```diff
@@ -1,12 +1,65 @@
-import React from "react";
+import { useState } from "react";
+import datajson from "../assets/CountriesData.json";
+import "../assets/css/common.css";
+import "../assets/css/main.css";
+import Navbar from "../components/ui-elements/Navbar";
+import DropDownFilter from "../components/filters/DropDownFilter";
+import InputFilter from "../components/filters/InputFilter";
+import Card from "../components/ui-elements/Card";
 
 const Home = () => {
-    return (
-        // TODO: Home page
-        // Render Country component (components/Country.jsx) for each country
-        // Take data from (assets/CountriesData.json)
-        <div>Home</div>
+  const [countries, setCountries] = useState(datajson);
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [selectedRegion, setSelectedRegion] = useState("All");
+
+  const handleSearch = (e) => {
+    const searchValue = e.target.value.toLowerCase();
+    const filteredCountries = datajson.filter((country) =>
+      country.name.toLowerCase().includes(searchValue)
+    );
+    setCountries(filteredCountries);
+  };
+
+  const showFilterByRegion = () => {
+    setIsDropdownOpen(!isDropdownOpen);
+  };
+
+  const filterByRegion = (region) => {
+    setSelectedRegion(region);
+
+    const filteredCountries = datajson.filter(
+      (country) => region === "All" || country.region === region
     );
+    setCountries(filteredCountries);
+  };
+
+  return (
+    <>
+      <Navbar />
+      <section className="filters">
+        <div className="container">
+          <InputFilter onChange={handleSearch} />
+          <DropDownFilter
+            onClick={showFilterByRegion}
+            isOpen={isDropdownOpen}
+            filterByRegion={filterByRegion}
+            region={selectedRegion}
+          />
+        </div>
+      </section>
+      <main className="main">
+        <div className="container">
+          <section>
+            <div className="countries-grid" id="countries-list">
+              {countries.map((country, index) => (
+                <Card key={index} country={country} />
```

Comment:
The `key` prop must be unique and static. 
Index depends on the arrays size. When we will do changes in this array (filter, delete, add) react will have a problem to know which element needs to re-render.